spek-junit-platform-engine/src/test/kotlin/org/jetbrains/spek/engine/LifecycleManagerTest.kt

```diff
@@ -0,0 +1,33 @@
+package org.jetbrains.spek.engine
+
+import org.jetbrains.spek.api.lifecycle.LifecycleListener
+import org.jetbrains.spek.engine.lifecycle.LifecycleManager
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import kotlin.properties.Delegates
+
+/**
+ * @author Ranie Jade Ramiso
+ */
+class LifecycleManagerTest {
+    var lifecycleManager by Delegates.notNull<LifecycleManager>()
+
+    @BeforeEach
+    fun setup() {
+        lifecycleManager = LifecycleManager()
+    }
+
+    @Test
+    fun sameListenerRegistration() {
+        val listener = object: LifecycleListener { }
+        lifecycleManager.addListener(listener)
+
+        try {
+            lifecycleManager.addListener(listener)
+            Assertions.fail("it should throw IllegalArgumentException")
+        } catch (e: IllegalArgumentException) {
+            // pass
```

Comment:
I'd also add check for error message, it's useful in case if in future we'll have several different cases when we'll need to throw `IAE`, up to u though :)